src/library_promise.js

```diff
@@ -21,6 +21,7 @@ mergeInto(LibraryManager.library, {
       promiseInfo.resolve = resolve;
     });
     promiseInfo.id = promiseMap.allocate(promiseInfo);
+    {{{ runtimeKeepalivePush() }}};
```

Comment:
There's no guarantee that exactly one callback will be installed on this new promise, so let's move the keepalive push to `emscripten_promise_then`. That will correctly handle the cases of 0 or multiple callbacks.

Reply:
Done